### DISCUSSION TITLE
What is `staleWhileRevalidate: true` in `withUrqlClient`?

### DISCUSSION THREAD

**Discussion by nghiepdev** [urql-graphql/urql#2447] [Q&A]
Hi,
In the https://formidable.com/open-source/urql/docs/advanced/server-side-rendering/#stale-while-revalidate
> In this case, we may want to update this data immediately after rehydration.

I still haven't really seen a difference when enabling `staleWhileRevalidate: true`.
As I understand: After rehydration, the client-side should be re-fetch to get the latest data, but I don't see any network call on the client.

**Comment by mikehaertl** [comment]
@kitten I have the same question. Could you maybe give a better explanation of what this feature is for? Isn't it exactly what you get if you **don't use ssrExchange** at all? I.e. the content is rendered on the server, sent to the client and the client then initiates another urql operation. Why would I use `ssrExchange` in the first place in this case?

Background: I tried to work on a fix for my problem #2689 which was introduced by a fix for `staleWhileRevalidate` which I need to understand first.

**Reply by kitten** [reply]
No, you can't not use `ssrExchange`, since the hydration wouldn't succeed, i.e. you would server-side render, then the client would be stuck in a differing loading state (or suspense state) on hydration.

That means that we need to expose a configuration option for this behaviour explicitly to render once with the stale data, then refetch data from the API: https://github.com/FormidableLabs/urql/blob/0582f9b138ddb66c5164b00a978aac90e865c0dc/packages/core/src/exchanges/ssr.ts#L146-L150

The intention here is for SSR results to be cacheable while still being able to refetch results from a backend, without coding a custom exchange.

**Reply by mikehaertl** [reply]
Thanks for the explanation.

> you would server-side render, then the client would be stuck in a differing loading state (or suspense state) on hydration

Could this be a React specific issue then? Because I haven't seen this in Vue / Nuxt 3. There it behaves like I said: Without `ssrExchange` there's a query during SSR on the server and another query after the page was rehydrated on the client. No broken state or similar.
